### PR TITLE
Datasets branch made into a SOAP WebService

### PR DESCRIPTION
**Pull request by Mchapuis**
We made temporary fixes to be transform the Generator into a web service. Please review the WebService_README.txt file in /doc folder for details.

This version is still not working correctly because we are back to our original problem. When I did the wrapper for the Generator, I used the swing library and java.io.File to save files. I need to correct this because JAX-RPC is not using these.

``The service class "ca.concordia.cse.gipsy.ws.soap.Generator" does not comply to one or more requirements of the JAX-RPC 1.1 specification, and may not deploy or function correctly.
  The method "getOutputFolder" on the service class "ca.concordia.cse.gipsy.ws.soap.Generator" uses a data type, "java.io.File", that is not supported by the JAX-RPC specification. Instances of the type may not serialize or deserialize correctly. Loss of data or complete failure of the Web service may result.
  The method "containsFile" on the service class "ca.concordia.cse.gipsy.ws.soap.Generator" uses a data type, "java.io.File", that is not supported by the JAX-RPC specification. Instances of the type may not serialize or deserialize correctly. Loss of data or complete failure of the Web ``

But the project is building correctly now.

### CODE REVIEW

**Comment by smokhov** [comment]
- [ ] @jdesorm1 -- please separate PM2 stuff from another branch and another PR, don't mix it into PM1

**Comment by smokhov** [comment]
@jashanj0tsingh -- please review PM1.3 portion of this PR -- comment on its description, diffs, etc. whether it adequately matches the spec; then test quickly a copy of it locally.

**Comment by smokhov** [comment]
- [ ]  (A) @jashanj0tsingh for example one of the concerns here is the large number of files changed -- are they all related to the Unicode character problems or something else

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+/bin
```
Comment (smokhov) [inline .gitignore:1]: - [ ]  (B) it seems this addition came in late; tons of stuff got committed into `bin/`; can we have it purged? Best way is to rewrite the history with `bin/` removed and force-push to the same branch.

**Review by jashanj0tsingh — COMMENTED** [review]
- [ ] (C) Since the libs have been present all along when WSC-Gen was first forked It is desirable to put all of them at one place so as to document these dependencies.

**Comment by smokhov** [comment]
- [ ] (D) @Mchapuis -- I don't follow why you need `java.io.File` in the web service; all you need in your service wrapper is a byte array (file contents) and a string (filename). These two fields can grouped in a serializable class say `ServiceFile` an that's what your WSDL methods return and the client side would adapt them back to an actual file, so you'd have an adapter class to help you with that in some util subpackage that both client and service could use.

File: .gitignore (hunk 1 above)
Comment (Mchapuis) [reply .gitignore:1]: I will try to do those changes as soon as I can

**Comment by Mchapuis** [comment]
about d) Yes, when @mariegobran showed me the error I understood that I had made a mistake. I need to modify the wrapper like you described as soon as I can for the rest of the PM's to work correctly.

**Review by jashanj0tsingh — CHANGES_REQUESTED** [review]
@Mchapuis The project as is does not get imported easily into OpenESB unless you delete `.nbproject` folder and import it as a `new` project with existing sources. Are you facing a similar problem? If no direct towards your documentation because I didn't find any except the `WebService_Readme.txt` in the `/doc` which merely describes the workaround of commenting out code. If there doesn't exist any please provide one.

**Comment by smokhov** [comment]
@jashanj0tsingh -- the project report and class diagram issues belong to the bb issues for now; no need for that part here in this repo. README, OpenESB, etc. are valid.

**Comment by Mchapuis** [comment]
@jashanj0tsingh We are facing similar problems when opening/importing the project in OpenESB. We still don't know the correct way to do it.

**Comment by smokhov** [comment]
@Mchapuis @jdesorm1 @mariegobran -- folks, this PR is about PM1 and its associated issues. For now please stop pushing PM2 stuff to your `Datatsets` branch as anything you push there in your fork ends up in this PR until it's merged. Please separate PM2 stuff into PM2 branch (off the tip of `Datasets`) and keep it there until it's time to submit. You can sync PM2 branch from Datasets from time to time but don't "litter" this PR else's it's going to delay further the review. Please move (git allows you to) the commits like a63bd6d, 433b2b3 to the PM2 branch in your fork.

**Comment by Mchapuis** [comment]
The push @mariegobran and me did yesterday was for PM1, but we should have referenced PM1 for clarity, we will do this next time.
@jdesorm1 fixed his push so now `Datasets` should be ok and contain only PM1 work.

**Comment by Mchapuis** [comment]
I don't know what happened with the `Datasets` branch but it seems like it has been rebase or something else happened. I am trying to fix this, we might have to revert and start over if I can't find the problem.

**Comment by Mchapuis** [comment]
I need to create a new pull request with the correct changes. This branch should be deleted.

**Comment by smokhov** [comment]
@Mchapuis -- I recommend against deleting the `Datasets` branch in your fork since we still have upstream. Instead, once your PR #7 is merged, rebase (force) your `Datasets` from us. And for the future really use branches like `PM2` off the current tip of whatever branch you are working off from.

**Pull request closed without merging**